sycl/test-e2e/Compat/helloworld.cpp

```diff
@@ -0,0 +1,136 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Part of the LLVM Project, under the Apache License v2.0 with LLVM
+ *  Exceptions. See https://llvm.org/LICENSE.txt for license information.
+ *  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCLcompat API
+ *
+ *  helloworld.cpp
+ *
+ *  Description:
+ *    Checks that the SYCLcompat example program compiles and runs
+ **************************************************************************/
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/sycl.hpp>
+#include <syclcompat/syclcompat.hpp>
+
+#include <cassert>
+#include <iostream>
+
+#include <sycl/sycl.hpp>
+
+/**
+ * Slope intercept form of a straight line equation: Y = m * X + b
+ */
+template <int BLOCK_SIZE>
+void slope_intercept(float *Y, float *X, float m, float b, size_t n) {
+
+  // Block index
+  size_t bx = syclcompat::work_group_id::x();
+  // Thread index
+  size_t tx = syclcompat::local_id::x();
+
+  size_t i = bx * BLOCK_SIZE + tx;
+  // or  i = syclcompat::global_id::x();
+  if (i < n)
+    Y[i] = m * X[i] + b;
+}
+
+void check_memory(void *ptr, std::string msg) {
+  if (ptr == nullptr) {
+    std::cerr << "Failed to allocate memory: " << msg << std::endl;
+    exit(EXIT_FAILURE);
+  }
+}
+
+/**
+ * Program main
+ */
+int main(int argc, char **argv) {
+  std::cout << "Simple Kernel example" << std::endl;
+
+  constexpr size_t n_points = 32;
+  constexpr float m = 1.5f;
+  constexpr float b = 0.5f;
+
+  int block_size = 32;
+  if (block_size > syclcompat::get_current_device()
+                       .get_info<sycl::info::device::max_work_group_size>())
+    block_size = 16;
```

Comment:
```suggestion
                       .get_info<sycl::info::device::max_work_group_size>()) {
    block_size = 16;
  }
```

I suppose we allow single line bodies without braces, but this particular line is a bit difficult to read.